### PR TITLE
Fix for #2 and #4, including some improvements

### PR DESCRIPTION
**Pull request by wucke13**
Hi,
I like the idea of this plugin, but I have some issues with it's functionality. I opened three issues regarding those (of which one is already closed), and this PR contains the fixes for the things I considered broken.
Obviously, some of the changes I did are more a question of preference than of correctness. I would be happy to discuss the changes with you (and have them merged eventually), as I do not want to maintain my own fork of this repo. Looking forward to the discussion!

### CODE REVIEW

**Comment by paretje** [comment]
Hi, thanks for your efforts!

Moving to `json` is indeed the right thing to do, not sure why I used the sexp format ...

Regarding the changes to the default search term: I guess it is better to move away from `tag:sent`, as this was already mentioned in #1 as well. I myself found it to be more natural to limit this to addresses you actually sent mails to, but I guess this is not the general consensus, and it is still mentioned in the README for those like me, who want to stick to the old behaviour. I wonder why `--deduplicate=address` was included in the README, but not in the actual default ...

I'll try your changes first, probably tomorrow, but at the moment, I don't see any reason not to merge them.

**Comment by paretje** [comment]
Looks good to me. Could you update the README regarding `--format=sexp` vs `--format=json` though?

**Comment by paretje** [comment]
Another remark: I'm not sure how this fixes #2. Isn't that simply controlled by `camel_case`, `ignore_case` and `smart_case` and the used matchers?

**Comment by wucke13** [comment]
Good to hear from you! I'm also unsure of #2. I ported the searching to `notmuch`'s search engine, before I found out about why it was the wrong thing to do according to `deoplete` guidelines (reference: #3). With that changes, the search results were better IMHO (and #2 was solved for that particular example). `notmuch` has a fairly advanced search engine built in (https://xapian.org/), and I'm still not convinced that it is the right thing __not to use__ it. Searching with it will be different though, when looking for `paretje` for example, searching for more than `par` will not popup `paretje@mail.com`, as `notmuch` doesn't try to fit small words in big words. I think that is clever, however, it will not comply to what most users would expect (especially, that the suggestion for `paretje@mail.com` vanishes once they type in the `e` of `pare` :laughing: ).

Regarding the README.md: I will add a commit to the PR in the next days, with added information.

**Comment by wucke13** [comment]
Ping @paretje 
I added the information to the README.md. Please let me know if there are any showstoppers left :smile:

File: README.md [hunk 1]
```diff
@@ -15,8 +15,22 @@ Plug 'paretje/deoplete-notmuch', {'for': 'mail'}
...
+[
+    {
+        "name":"Jane Doe",
+        "address":"jane@mail.doe",
+        "name-addr":"Jane Doe <jane@mail.doe>"
+    },
+]
 ```
+
+You can modify the `notmuch search` to great extents, for example replace `'*'` by `tag:sent` to narrow the completions down to people who already received an email from you. For further information, see `man 1 notmuch-address` and `man 7 notmuch-search-terms`.
```
Comment (paretje) [inline README.md]: One small remark: `tag:sent` also requires the option `--output=recipients`.

**Review by paretje — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (wucke13) [reply README.md]: Thanks, should be resolved now

**Comment by paretje** [comment]
@wucke13: thanks for the contribution!

**Comment by wucke13** [comment]
You're welcome!

**Pull request merged**